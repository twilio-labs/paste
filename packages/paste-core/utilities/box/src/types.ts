import * as CSS from 'csstype';
import {ThemeShape} from '@twilio-paste/theme-tokens';

export interface BackgroundColorProp {
  backgroundColor?: keyof ThemeShape['backgroundColors'];
}

export interface BorderColorProps {
  borderColor?: keyof ThemeShape['borderColors'];
  borderBottomColor?: keyof ThemeShape['borderColors'];
  borderLeftColor?: keyof ThemeShape['borderColors'];
  borderRightColor?: keyof ThemeShape['borderColors'];
  borderTopColor?: keyof ThemeShape['borderColors'];
}

export interface BorderRadiusProps {
  borderRadius?: keyof ThemeShape['radii'];
  borderBottomLeftRadius?: keyof ThemeShape['radii'];
  borderBottomRightRadius?: keyof ThemeShape['radii'];
  borderTopLeftRadius?: keyof ThemeShape['radii'];
  borderTopRightRadius?: keyof ThemeShape['radii'];
}

export interface BorderStyleProp {
  borderStyle?: CSS.BorderStyleProperty;
}

export interface BorderWidthProps {
  borderWidth?: keyof ThemeShape['borderWidths'];
  borderBottomWidth?: keyof ThemeShape['borderWidths'];
  borderLeftWidth?: keyof ThemeShape['borderWidths'];
  borderRightWidth?: keyof ThemeShape['borderWidths'];
  borderTopWidth?: keyof ThemeShape['borderWidths'];
}

export interface HeightProp {
  height?: keyof ThemeShape['heights'];
}

export interface MarginProps {
  m?: keyof ThemeShape['space'];
  margin?: keyof ThemeShape['space'];
  mt?: keyof ThemeShape['space'];
  marginTop?: keyof ThemeShape['space'];
  mr?: keyof ThemeShape['space'];
  marginRight?: keyof ThemeShape['space'];
  mb?: keyof ThemeShape['space'];
  marginBottom?: keyof ThemeShape['space'];
  ml?: keyof ThemeShape['space'];
  marginLeft?: keyof ThemeShape['space'];
  mx?: keyof ThemeShape['space'];
  my?: keyof ThemeShape['space'];
}

export interface MaxWidthProp {
  maxWidth?: keyof ThemeShape['maxWidths'];
}

export interface MinWidthProp {
  minWidth?: keyof ThemeShape['maxWidths'];
}

export interface PaddingProps {
  p?: keyof ThemeShape['space'];
  padding?: keyof ThemeShape['space'];
  pt?: keyof ThemeShape['space'];
  paddingTop?: keyof ThemeShape['space'];
  pr?: keyof ThemeShape['space'];
  paddingRight?: keyof ThemeShape['space'];
  pb?: keyof ThemeShape['space'];
  paddingBottom?: keyof ThemeShape['space'];
  pl?: keyof ThemeShape['space'];
  paddingLeft?: keyof ThemeShape['space'];
  px?: keyof ThemeShape['space'];
  py?: keyof ThemeShape['space'];
}

export interface SpacingThemeProps extends MarginProps, PaddingProps {}

export interface WidthProps {
  width?: keyof ThemeShape['widths'];
}
