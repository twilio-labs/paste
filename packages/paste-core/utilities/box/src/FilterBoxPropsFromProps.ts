import * as lodash from 'lodash';

const supportedBoxProps = [
  'margin',
  'm',
  'marginTop',
  'mt',
  'marginRight',
  'mr',
  'marginBottom',
  'mb',
  'marginLeft',
  'ml',
  'marginX',
  'mx',
  'marginY',
  'my',
  'padding',
  'p',
  'paddingTop',
  'pt',
  'paddingRight',
  'pr',
  'paddingBottom',
  'pb',
  'paddingLeft',
  'pl',
  'paddingX',
  'px',
  'paddingY',
  'py',
  'width',
  'height',
  'display',
  'minWidth',
  'minHeight',
  'maxWidth',
  'maxHeight',
  'size',
  'verticalAlign',
  'overflow',
  'overflowX',
  'overflowY',
  'alignItems',
  'alignContent',
  'justifyItems',
  'justifyContent',
  'flexWrap',
  'flexDirection',
  'flex',
  'flexGrow',
  'flexShrink',
  'flexBasis',
  'justifySelf',
  'alignSelf',
  'order',
  'backgroundImage',
  'backgroundSize',
  'backgroundPosition',
  'backgroundRepeat',
  'backgroundColor',
  'border',
  'borderWidth',
  'borderStyle',
  'borderColor',
  'borderRadius',
  'borderTop',
  'borderTopWidth',
  'borderTopStyle',
  'borderTopColor',
  'borderTopLeftRadius',
  'borderTopRightRadius',
  'borderRight',
  'borderRightWidth',
  'borderRightStyle',
  'borderRightColor',
  'borderBottom',
  'borderBottomWidth',
  'borderBottomStyle',
  'borderBottomColor',
  'borderBottomLeftRadius',
  'borderBottomRightRadius',
  'borderLeft',
  'borderLeftWidth',
  'borderLeftStyle',
  'borderLeftColor',
  'borderX',
  'borderY',
  'borderColor',
  'boxShadow',
  'zIndex',
] as const;

type Exceptions = typeof supportedBoxProps | string[];

export const filterBoxPropsFromProps = (props: {}, exceptions: Exceptions = []): {} => {
  const boxPropsToFilter = supportedBoxProps.filter(prop => {
    return !exceptions.includes(prop);
  });
  return lodash.omit(props, boxPropsToFilter);
};
