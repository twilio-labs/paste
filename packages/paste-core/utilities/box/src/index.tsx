import styled from '@emotion/styled';
import {
  display,
  DisplayProps,
  borders,
  BorderProps,
  height,
  maxWidth,
  minWidth,
  space,
  style,
  width,
} from 'styled-system';

import {
  SpacingThemeProps,
  BackgroundColorProp,
  BorderRadiusProps,
  BorderWidthProps,
  WidthProps,
  HeightProp,
  MaxWidthProp,
  MinWidthProp,
  BorderColorProps,
  BorderStyleProp,
} from './types';

export interface BoxProps
  extends DisplayProps,
    BorderProps,
    HeightProp,
    MaxWidthProp,
    MinWidthProp,
    WidthProps,
    SpacingThemeProps,
    BackgroundColorProp,
    BorderColorProps,
    BorderRadiusProps,
    BorderStyleProp,
    BorderWidthProps {
  as?: keyof JSX.IntrinsicElements;
}

const backgroundColor = style({
  prop: 'backgroundColor',
  cssProperty: 'backgroundColor',
  key: 'backgroundColors',
});

const borderBottomLeftRadius = style({
  prop: 'borderBottomLeftRadius',
  cssProperty: 'borderBottomLeftRadius',
  key: 'radii',
});

const borderBottomRightRadius = style({
  prop: 'borderBottomRightRadius',
  cssProperty: 'borderBottomRightRadius',
  key: 'radii',
});

const borderTopLeftRadius = style({
  prop: 'borderTopLeftRadius',
  cssProperty: 'borderTopLeftRadius',
  key: 'radii',
});

const borderTopRightRadius = style({
  prop: 'borderTopRightRadius',
  cssProperty: 'borderTopRightRadius',
  key: 'radii',
});

const borderColor = style({
  prop: 'borderColor',
  cssProperty: 'borderColor',
  key: 'borderColors',
});

const borderBottomWidth = style({
  prop: 'borderBottomWidth',
  cssProperty: 'borderBottomWidth',
  key: 'borderWidths',
});

const borderLeftWidth = style({
  prop: 'borderLeftWidth',
  cssProperty: 'borderLeftWidth',
  key: 'borderWidths',
});

const borderRightWidth = style({
  prop: 'borderRightWidth',
  cssProperty: 'borderRightWidth',
  key: 'borderWidths',
});

const borderTopWidth = style({
  prop: 'borderTopWidth',
  cssProperty: 'borderTopWidth',
  key: 'borderWidths',
});

const Box = styled.div<BoxProps>`
  ${backgroundColor};
  ${borders};
  ${borderColor}
  ${borderBottomWidth}
  ${borderLeftWidth}
  ${borderRightWidth}
  ${borderTopWidth}
  ${borderBottomLeftRadius};
  ${borderBottomRightRadius};
  ${borderTopLeftRadius};
  ${borderTopRightRadius};
  ${display};
  ${height};
  ${maxWidth};
  ${minWidth};
  ${space};
  ${width};
`;

Box.defaultProps = {
  minWidth: 'size0',
};

export {Box};
