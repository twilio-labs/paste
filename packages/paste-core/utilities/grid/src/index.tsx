export * from './Grid';
export * from './Column';
