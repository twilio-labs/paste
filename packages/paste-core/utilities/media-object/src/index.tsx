import styled from '@emotion/styled';
import {space, Theme} from 'styled-system';
import {themeGet} from '@styled-system/theme-get';
import {SpaceProps} from '@twilio-paste/style-props';

export interface MediaObjectProps extends Theme {
  isCentered?: boolean;
}

const MediaObject = styled.div<MediaObjectProps>`
  align-items: ${(props: MediaObjectProps) => (props.isCentered ? 'center' : 'flex-start')};
  display: flex;
`;

export interface MediaFigureProps extends Theme {
  align?: 'start' | 'end';
  spacing?: 'space20' | 'space30';
}

const getMargin = (props: MediaFigureProps): string => {
  const direction = props.align === 'end' ? 'left' : 'right';
  const spacing = themeGet(`space.${props.spacing}`)(props);
  return `margin-${direction}: ${spacing}`;
};

const MediaFigure = styled.div<MediaFigureProps>`
  flex-shrink: 0;
  ${getMargin};
`;

MediaFigure.defaultProps = {
  align: 'start',
  spacing: 'space20',
};

export interface MediaBodyProps extends Theme, SpaceProps {}

const MediaBody = styled.div<MediaBodyProps>`
  flex: 1;
  min-width: ${themeGet('minWidths.size0')};
  ${space};
`;

const Media = {
  Object: MediaObject,
  Figure: MediaFigure,
  Body: MediaBody,
};

export {Media};
