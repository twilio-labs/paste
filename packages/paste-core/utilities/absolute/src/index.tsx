import styled from '@emotion/styled';
import {top, right, bottom, left, zIndex, TopProps, BottomProps, LeftProps, RightProps, styleFn} from 'styled-system';
import {Box, BoxProps} from '@twilio-paste/box';
import {ThemeShape} from '@twilio-paste/theme-tokens';

const LEFT_OPTIONS = [
  'fill',
  'left',
  'left_top',
  'left_bottom',
  'left_fill',
  'top_left',
  'top_fill',
  'bottom_left',
  'bottom_fill',
];
const RIGHT_OPTIONS = [
  'fill',
  'right',
  'right_top',
  'right_bottom',
  'right_fill',
  'top_right',
  'top_fill',
  'bottom_right',
  'bottom_fill',
];
const TOP_OPTIONS = [
  'fill',
  'top',
  'top_right',
  'top_left',
  'top_fill',
  'right_top',
  'right_fill',
  'left_top',
  'left_fill',
];
const BOTTOM_OPTIONS = [
  'fill',
  'bottom',
  'bottom_right',
  'bottom_left',
  'bottom_fill',
  'right_bottom',
  'right_fill',
  'left_bottom',
  'left_fill',
];

export type Presets =
  | 'fill'
  | 'top'
  | 'top_left'
  | 'top_right'
  | 'top_fill'
  | 'bottom'
  | 'bottom_left'
  | 'bottom_right'
  | 'bottom_fill'
  | 'left'
  | 'left_top'
  | 'left_bottom'
  | 'left_fill'
  | 'right'
  | 'right_top'
  | 'right_bottom'
  | 'right_fill';

export interface ZIndexProps {
  zIndex?: keyof ThemeShape['zIndices'];
}

export interface AbsoluteProps extends TopProps, BottomProps, LeftProps, RightProps, ZIndexProps, BoxProps {
  preset?: Presets;
}

interface PositionStyles {
  top?: 0;
  right?: 0;
  bottom?: 0;
  left?: 0;
}

const preset: styleFn = ({preset: _preset}) => {
  const positionStyles: PositionStyles = {};

  if (_preset == null) {
    return positionStyles;
  }

  if (LEFT_OPTIONS.includes(_preset)) {
    positionStyles.left = 0;
  }
  if (RIGHT_OPTIONS.includes(_preset)) {
    positionStyles.right = 0;
  }
  if (TOP_OPTIONS.includes(_preset)) {
    positionStyles.top = 0;
  }
  if (BOTTOM_OPTIONS.includes(_preset)) {
    positionStyles.bottom = 0;
  }
  return positionStyles;
};

const Absolute = styled(Box)<AbsoluteProps>(preset, zIndex, top, right, bottom, left, {
  position: 'absolute',
});

Absolute.defaultProps = {
  preset: 'fill',
};

export {Absolute};
