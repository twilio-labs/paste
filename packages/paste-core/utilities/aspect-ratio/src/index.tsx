import * as React from 'react';
import styled from '@emotion/styled';
import {Absolute} from '@twilio-paste/absolute';

interface AspectRatioProps {
  ratio: string;
  children: NonNullable<React.ReactNode>;
}

const RATIO_REGEX = /^(\d+:\d*)$/;

const isCorrectPattern = (ratio: string): boolean => RATIO_REGEX.test(ratio);

const handlePropValidation = ({ratio}: AspectRatioProps): void => {
  const hasRatio = ratio != null && ratio !== '';

  if (!hasRatio) {
    throw new Error(`[Paste: AspectRatio] Missing 'ratio' prop.`);
  }

  if (!isCorrectPattern(ratio)) {
    throw new Error(`[Paste: AspectRatio] 'ratio' is invalid. Use a colon separated number pattern (4:3).`);
  }
};

const AspectRatioContainer = styled.div`
  position: relative;
`;

const AspectRatio: React.FC<AspectRatioProps> = props => {
  handlePropValidation(props);

  const aspectArray = props.ratio.split(':').map(Number);
  const aspectPercent = (aspectArray[1] / aspectArray[0]) * 100;

  return (
    <AspectRatioContainer style={{paddingBottom: `${aspectPercent}%`}}>
      <Absolute>{props.children}</Absolute>
    </AspectRatioContainer>
  );
};

AspectRatio.defaultProps = {
  ratio: '4:3',
};

export {AspectRatio};
