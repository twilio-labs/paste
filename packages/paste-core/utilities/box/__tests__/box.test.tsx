import * as React from 'react';
import renderer from 'react-test-renderer';
import {Theme} from '@twilio-paste/theme';
import {Box} from '../src';

describe('Backgrounds', () => {
  it('it should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box backgroundColor="colorBackgroundPrimary">background single</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should render responsive values', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box backgroundColor={['colorBackgroundPrimaryLight', 'colorBackgroundPrimary']}>background responsive</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Borders', () => {
  it('it should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box
            borderStyle="solid"
            borderColor="colorBorderPrimaryDark"
            borderWidth="borderWidth10"
            borderRadius="borderRadius20"
          >
            border single
          </Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should render responsive values', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box
            borderStyle={['dashed', 'dotted', 'solid']}
            borderColor={['colorBorderPrimaryDark', 'colorBorderPrimaryLight']}
            borderWidth={['borderWidth10', 'borderWidth20']}
            borderRadius={['borderRadius0', 'borderRadius10']}
          >
            border responsive
          </Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Sizes', () => {
  it('it should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box width="size10" minWidth="size0" maxWidth="size20" height="size10" minHeight="size0" maxHeight="size20">
            size single
          </Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should render responsive values', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box
            width={['size10', 'size20']}
            minWidth={['size0', 'size10']}
            maxWidth={['size20', 'size30']}
            height={['size10', 'size20']}
            minHeight={['size0', 'size10']}
            maxHeight={['size20', 'size30']}
          >
            size responsive
          </Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Spaces', () => {
  it('(A) it should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box margin="space20">space single 1</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('(A) it should render responsive values', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box margin={['space20', 'space30']}>space responsive 1</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('(B) it should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box marginTop="space20" marginRight="space20" marginBottom="space30" marginLeft="space30">
            space single 2
          </Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('(B)it should render responsive values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box
            marginTop={['space20', 'space30']}
            marginRight={['space20', 'space30']}
            marginBottom={['space30', 'space40']}
            marginLeft={['space30', 'space40']}
          >
            space responsive 2
          </Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Shadows', () => {
  it('it should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box boxShadow="shadowCard">shadow single</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should render responsive values', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box boxShadow={['shadowCard', 'shadowFocus']}>shadow responsive</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ZIndex', () => {
  it('it should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box zIndex="zIndex10">z-index single</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('it should render responsive values', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box zIndex={['zIndex10', 'zIndex20']}>z-index responsive</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
