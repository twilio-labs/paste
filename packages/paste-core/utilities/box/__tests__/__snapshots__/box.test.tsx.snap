// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backgrounds it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  background-color: #b3d3e9;
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    background-color: #027ac5;
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    background responsive
  </div>
</div>
`;

exports[`Backgrounds it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  background-color: #027ac5;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    background single
  </div>
</div>
`;

exports[`Borders it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  border-style: dashed;
  border-color: #003e82;
  border-width: 1px;
  border-radius: 0;
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    border-style: dotted;
    border-color: #b3d3e9;
    border-width: 2px;
    border-radius: 3px;
  }
}

@media screen and (min-width:64rem) {
  .emotion-0 {
    border-style: solid;
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    border responsive
  </div>
</div>
`;

exports[`Borders it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  border-style: solid;
  border-color: #003e82;
  border-width: 1px;
  border-radius: 4px;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    border single
  </div>
</div>
`;

exports[`Shadows it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  box-shadow: 0 2px 4px 0 rgba(40,42,43,0.3);
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    box-shadow: 0 0 0 4px rgba(2,122,197,0.8);
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    shadow responsive
  </div>
</div>
`;

exports[`Shadows it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  box-shadow: 0 2px 4px 0 rgba(40,42,43,0.3);
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    shadow single
  </div>
</div>
`;

exports[`Sizes it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  width: 5.5rem;
  min-width: 0;
  max-width: 12rem;
  height: 5.5rem;
  min-height: 0;
  max-height: 12rem;
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    width: 12rem;
    min-width: 5.5rem;
    max-width: 18.5rem;
    height: 12rem;
    min-height: 5.5rem;
    max-height: 18.5rem;
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
    height={
      Array [
        "size10",
        "size20",
      ]
    }
    width={
      Array [
        "size10",
        "size20",
      ]
    }
  >
    size responsive
  </div>
</div>
`;

exports[`Sizes it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  width: 5.5rem;
  min-width: 0;
  max-width: 12rem;
  height: 5.5rem;
  min-height: 0;
  max-height: 12rem;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
    height="size10"
    width="size10"
  >
    size single
  </div>
</div>
`;

exports[`Spaces (A) it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  margin: 0.25rem;
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    margin: 0.5rem;
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    space responsive 1
  </div>
</div>
`;

exports[`Spaces (A) it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  margin: 0.25rem;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    space single 1
  </div>
</div>
`;

exports[`Spaces (B) it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  margin-top: 0.25rem;
  margin-right: 0.25rem;
  margin-bottom: 0.5rem;
  margin-left: 0.5rem;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    space single 2
  </div>
</div>
`;

exports[`Spaces (B)it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  margin-top: 0.25rem;
  margin-right: 0.25rem;
  margin-bottom: 0.5rem;
  margin-left: 0.5rem;
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    margin-top: 0.5rem;
    margin-right: 0.5rem;
    margin-bottom: 0.75rem;
    margin-left: 0.75rem;
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    space responsive 2
  </div>
</div>
`;

exports[`ZIndex it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  z-index: 10;
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    z-index: 20;
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    z-index responsive
  </div>
</div>
`;

exports[`ZIndex it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  z-index: 10;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    z-index single
  </div>
</div>
`;
